python/lsst/scarlet/lite/image.py

```diff
@@ -0,0 +1,1372 @@
+    of ``bands``:
+
+    >>> image = Image.from_box(Box((3, 4)), bands=("r", "i"))
+    >>> print(image)
+    Image:
+     [[[0. 0. 0. 0.]
+      [0. 0. 0. 0.]
+      [0. 0. 0. 0.]]
+    <BLANKLINE>
+     [[0. 0. 0. 0.]
+      [0. 0. 0. 0.]
+      [0. 0. 0. 0.]]]
+      bands=('r', 'i')
+      bbox=<Box shape=(3, 4), origin=(0, 0)>
+
+    To select a sub-image use a ``Box`` to select a spatial region in either a
+    single-band or multi-band image:
+
+    >>> x = np.arange(60).reshape(3, 4, 5)
+    >>> image = Image(x, bands=("g", "r", "i"), yx0=(20, 30))
+    >>> bbox = Box((2, 2), (21, 32))
+    >>> print(image[bbox])
+    Image:
+     [[[ 7  8]
+      [12 13]]
+    <BLANKLINE>
+     [[27 28]
+      [32 33]]
+    <BLANKLINE>
+     [[47 48]
+      [52 53]]]
+      bands=('g', 'r', 'i')
+      bbox=<Box shape=(2, 2), origin=(21, 32)>
+
+
+    To select a single-band image from a muli-band image,
+    pass the name of the band as an index:
+
+    >>> print(image["r"])
+    Image:
+     [[20 21 22 23 24]
+     [25 26 27 28 29]
+     [30 31 32 33 34]
+     [35 36 37 38 39]]
+      bands=()
+      bbox=<Box shape=(4, 5), origin=(20, 30)>
+
+    Multi-band images can also be slcied in the spatial dimension, for example
+
+    >>> print(image["g":"r"])
+    Image:
+     [[[ 0  1  2  3  4]
+      [ 5  6  7  8  9]
+      [10 11 12 13 14]
+      [15 16 17 18 19]]
+    <BLANKLINE>
+     [[20 21 22 23 24]
+      [25 26 27 28 29]
+      [30 31 32 33 34]
+      [35 36 37 38 39]]]
+      bands=('g', 'r')
+      bbox=<Box shape=(4, 5), origin=(20, 30)>
+
+    and
+
+    >>> print(image["r":"r"])
+    Image:
+     [[[20 21 22 23 24]
+      [25 26 27 28 29]
+      [30 31 32 33 34]
+      [35 36 37 38 39]]]
+      bands=('r',)
+      bbox=<Box shape=(4, 5), origin=(20, 30)>
+
+    both extract a slice of a multi-band image.
+
+    .. warning::
+        Unlike numerical indices, where ``slice(x, y)`` will select the
+        subset of an array from ``x`` to ``y-1`` (excluding ``y``),
+        a spectral slice of an ``Image`` will return the image slice
+        including band ``y``.
+
+    It is also possible to change the order or index a subset of bands
+    in an image. For example:
+
+    >>> print(image[("r", "g", "i")])
+    Image:
+     [[[20 21 22 23 24]
+      [25 26 27 28 29]
+      [30 31 32 33 34]
+      [35 36 37 38 39]]
+    <BLANKLINE>
+     [[ 0  1  2  3  4]
+      [ 5  6  7  8  9]
+      [10 11 12 13 14]
+      [15 16 17 18 19]]
+    <BLANKLINE>
+     [[40 41 42 43 44]
+      [45 46 47 48 49]
+      [50 51 52 53 54]
+      [55 56 57 58 59]]]
+      bands=('r', 'g', 'i')
+      bbox=<Box shape=(4, 5), origin=(20, 30)>
+
+
+    will return a new image with the bands re-ordered.
+
+    Images can be combined using the standard arithmetic operations similar to
+    numpy arrays, including ``+, -, *, /, **`` etc, however, if two images are
+    combined with different bounding boxes, the _union_ of the two
```

Comment:
I see your point, but I want to try and have as little meta data as possible that would have to be persisted when images are combined. There are convenience methods that do allow for the intersection to be used, but I made union the default because, at least in scarlet, that is the desired behavior. For example, a source is made up of components that might have different bounding boxes, and a blend is made of of sources with different bounding boxes, but it's convenient to be able to just add them to create the full model.